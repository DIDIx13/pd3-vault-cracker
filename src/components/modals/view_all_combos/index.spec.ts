import '@testing-library/jest-dom';
import { render } from '@testing-library/svelte';

import Component from './index.svelte';
import { writable } from 'svelte/store';

test('should render without crashing', () => {
  const { container } = render(Component);
  expect(container).toMatchSnapshot();
});

test('should render with props', () => {
  const { container } = render(Component, {
    props: {
      combinations: [
        [6, 6, 6, 9],
        [6, 6, 9, 6],
        [6, 6, 9, 9],
        [6, 9, 6, 6],
        [6, 9, 6, 9],
        [6, 9, 9, 6],
        [6, 9, 9, 9],
        [9, 6, 6, 6],
        [9, 6, 6, 9],
        [9, 6, 9, 6],
        [9, 6, 9, 9],
        [9, 9, 6, 6],
        [9, 9, 6, 9],
        [9, 9, 9, 6],
      ],
    },
  });
  expect(container).toMatchSnapshot();
});

test('should print known fingerprints in header', () => {
  const pressedNumbers = writable([]);
  pressedNumbers.set([6, 9] as never[]);

  let knownFingerprints;

  pressedNumbers.subscribe((value) => {
    knownFingerprints = value.join('');
  });

  const { container } = render(Component, {
    props: {
      combination: pressedNumbers,
    },
  });

  // Expect All combinations for x
  expect(container).toHaveTextContent(
    `All combinations for ${knownFingerprints}`,
  );
});

test('should print a button for each combination', () => {
  const combinations = [
    [6, 6, 6, 9],
    [6, 6, 9, 6],
    [6, 6, 9, 9],
    [6, 9, 6, 6],
    [6, 9, 6, 9],
    [6, 9, 9, 6],
    [6, 9, 9, 9],
    [9, 6, 6, 6],
    [9, 6, 6, 9],
    [9, 6, 9, 6],
    [9, 6, 9, 9],
    [9, 9, 6, 6],
    [9, 9, 6, 9],
    [9, 9, 9, 6],
  ];

  const { container } = render(Component, {
    props: {
      combinations,
    },
  });

  // Expect 14 buttons
  expect(container.querySelectorAll('button')).toHaveLength(14);

  // Expect each button to have the correct text
  const buttons = container.querySelectorAll('button');
  for (const [index, button] of buttons.entries()) {
    expect(button).toHaveTextContent(`${[combinations][0][index].join('')}`);
  }
});
