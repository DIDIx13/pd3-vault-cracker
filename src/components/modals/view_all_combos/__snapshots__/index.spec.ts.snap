// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render with props 1`] = `
<body>
  <div>
    <dialog>
      <article>
        <header
          class="svelte-1hk3v5y"
        >
          <a
            aria-label="Close"
            class="close"
            href="#close"
          />
           
          All combinations for
           
          
        </header>
         
        <p
          class="svelte-1hk3v5y"
        >
          Left-click on a combination to mark it as tried
        </p>
         
        <p
          class="svelte-1hk3v5y"
        >
          Right-click/long-press on a combination to mark it as correct
        </p>
         
        <br />
         
        <div
          class="grid combo"
        >
          <button
            class="combo-btn btn-inverted svelte-1hk3v5y"
          >
            6669
             
          </button>
          <button
            class="combo-btn btn-inverted svelte-1hk3v5y"
          >
            6696
             
          </button>
          <button
            class="combo-btn btn-inverted svelte-1hk3v5y"
          >
            6699
             
          </button>
          <button
            class="combo-btn btn-inverted svelte-1hk3v5y"
          >
            6966
             
          </button>
          <button
            class="combo-btn btn-inverted svelte-1hk3v5y"
          >
            6969
             
          </button>
          <button
            class="combo-btn btn-inverted svelte-1hk3v5y"
          >
            6996
             
          </button>
          <button
            class="combo-btn btn-inverted svelte-1hk3v5y"
          >
            6999
             
          </button>
          <button
            class="combo-btn btn-inverted svelte-1hk3v5y"
          >
            9666
             
          </button>
          <button
            class="combo-btn btn-inverted svelte-1hk3v5y"
          >
            9669
             
          </button>
          <button
            class="combo-btn btn-inverted svelte-1hk3v5y"
          >
            9696
             
          </button>
          <button
            class="combo-btn btn-inverted svelte-1hk3v5y"
          >
            9699
             
          </button>
          <button
            class="combo-btn btn-inverted svelte-1hk3v5y"
          >
            9966
             
          </button>
          <button
            class="combo-btn btn-inverted svelte-1hk3v5y"
          >
            9969
             
          </button>
          <button
            class="combo-btn btn-inverted svelte-1hk3v5y"
          >
            9996
             
          </button>
        </div>
      </article>
    </dialog>
  </div>
</body>
`;

exports[`should render without crashing 1`] = `
<body>
  <div>
    <dialog>
      <article>
        <header
          class="svelte-1hk3v5y"
        >
          <a
            aria-label="Close"
            class="close"
            href="#close"
          />
           
          All combinations for
           
          
        </header>
         
        <p
          class="svelte-1hk3v5y"
        >
          Left-click on a combination to mark it as tried
        </p>
         
        <p
          class="svelte-1hk3v5y"
        >
          Right-click/long-press on a combination to mark it as correct
        </p>
         
        <br />
         
        <p
          class="svelte-1hk3v5y"
        >
          No combos found.
        </p>
      </article>
    </dialog>
  </div>
</body>
`;
